demisto_sdk/commands/common/singleton.py

```diff
@@ -1,6 +1,16 @@
 from abc import abstractmethod
 
 
+class SingletonMeta(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(SingletonMeta, cls).__call__(*args, **kwargs)
+
+        return cls._instances[cls]
```

Comment:
this saves multiple instances based on their args? why do we need a list of instances and not only a single one?

Reply:
Since this is a metaclass that can be used for multiple different classes across the SDK.

For example, if we have:
```python
class A(metaclass=SingletonMeta):
    pass

class B(metaclass=SingletonMeta):
    pass
```

After both have been created once, the `_instances` dict will contain `{A: <A class instance>, B: <B class instance>`.
Then when you try to create a new `A` / `B` instance, it will fetch the existing instance from the dict instead of creating a new one.

This is exact code is quite common, if you'll look up "Python Singleton implementation".